test/unit/pool/test-destroy-connection.js

```diff
@@ -11,10 +11,10 @@ server.listen(common.fakeServerPort, function (err) {
   pool.getConnection(function (err, connection) {
     assert.ifError(err);
 
-    assert.strictEqual(connection, pool._allConnections[0]);
+    assert.strictEqual(connection, pool._connectionManager._allConnections._map[1]);
```

Comment:
At this point in the module, probably every single property is important in some way. I would err on the site of just not changing them if you don't really need to.